wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,191 @@
+import django_filters
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db.models import Count, OuterRef, Q, Subquery
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.views.reports import ReportView
+from wagtail.coreutils import get_content_languages
+from wagtail.models import ContentType, Page, PageLogEntry, Site, get_page_models
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _get_site_choices():
+    choices = [(site.pk, str(site)) for site in Site.objects.all()]
+    return choices
+
+
+def _annotate_last_edit_info(queryset):
+    latest_edit_log = PageLogEntry.objects.filter(content_type=OuterRef("pk")).order_by(
+        "-timestamp", "-pk"
+    )
+
+    queryset = queryset.annotate(
+        count=Count("pages"),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+def _annotate_last_edit_info_by_locale(queryset, language_code):
+    latest_edit_log = PageLogEntry.objects.filter(
+        content_type=OuterRef("pk"), page__locale__language_code=language_code
+    ).order_by("-timestamp", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(pages__locale__language_code=language_code)),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
```

Comment:
Is there a way to come at this from the `User` model instead?

Reply:
Same thing as above, [can't do the reverse lookup](https://github.com/wagtail/wagtail/blob/85a4a9ebf744c4f4881744a26f3e48262da2543e/wagtail/models/audit_log.py#L175-L182).

Reply:
This one is more understandable because the field is defined on an abstract model. Last time I checked, there still wasn't a good solution for this (other than conpletely redefining the field on the concrete sublasses).

Reply:
Replacing with `Page.owner` would make these points redundant, I guess? I know it's very different to 'last_edited_by', but potentially more useful in some contexts, and certainly cheaper to fetch

Reply:
Done!